### DISCUSSION TITLE
Install older versions of ArchR

### DISCUSSION THREAD

**Discussion by RegnerM2015** [GreenleafLab/ArchR#656] [Questions / Documentation]
Hi @jgranja24 and @rcorces 
Has any one been able to install  ArchR_0.9.5? 

I recently did an analysis using  ArchR_0.9.5 and came back to re-run the analysis with Version 1.0.1. However, the number of cells processed in the Arrow File and ArchR project object are somehow different between versions 0.9.5 and 1.0.1.

I would like to revert back for reproducibility. Thanks for your help in advance!

**Comment by rcorces** [comment]
Unfortunately we were not implementing stable release tags back in beta (0.x.x) so I dont think there is a great way to do this. I just restored this branch https://github.com/GreenleafLab/ArchR/tree/dev_053120 which corresponds to last update of 0.9.5. I think you should be able to install this with devtools by specifying the branch name in the `ref` param. Can you let me know if that works?

Also, there was some issue with an off-by-one error on fragment starts or stops that was fixed in a later version of ArchR and I think that fix may have also introduced some differences in Arrow files.

devtools::install_github("GreenleafLab/ArchR", ref="dev_053120", repos = BiocManager::repositories())

**Comment by RegnerM2015** [comment]
@rcorces 

Thank you SO MUCH for your help. I was able to reinstall  ArchR_0.9.5 with:
devtools::install_github("GreenleafLab/ArchR", ref="dev_053120", repos = BiocManager::repositories())
Now I am able to generate Arrow files with the same number of barcodes as my original workflow.

I guess my follow up question is how important it is acknowledge the "off-by-one error on fragment starts or stops"? I have results in a paper that is currently under peer review. The difference in number of cells was pretty substantial compared to ArchR_0.9.5. Do you think this off-by-one error would cause ArchR to generate Arrow files with **_less_** barcodes? 

I don't think the downstream results would be affected, besides the barcode/cell number. How do you recommend I should proceed during the review process?

**Reply by rcorces** [reply]
I'm a little surprised to hear that there are big differences in the number of barcodes. I only mentioned the off-by-one issue as a warning on the absolute accuracy of the results. I do not think that any conclusions made from 0.9.5 would be affected by the off-by-one error except for maybe single-site footprinting which nobody is doing. I dont think this is important to acknowledge.

I dont think I know how to answer this. How substantial is the difference in the number of cells between 0.9.5 and 1.0.1? Anything about the cells that are different that gives a clue what is going on?

As far as I can tell, these two commits show the only changes to `createArrowFiles()` since 0.9.5:

https://github.com/GreenleafLab/ArchR/commit/ecfe12a6b4df80beec1bf16be3b398433a7742bd#diff-bab2afba66a9980d18aaaa80450eaeff5c80a12eb8a943c0680d7b71b06b7e5e

https://github.com/GreenleafLab/ArchR/commit/ecfe12a6b4df80beec1bf16be3b398433a7742bd#diff-bab2afba66a9980d18aaaa80450eaeff5c80a12eb8a943c0680d7b71b06b7e5e

**Reply by RegnerM2015** [reply]
At this point, it is difficult to compare these cell numbers directly, since I made the comparison in cell numbers after I did barcode QC and removal. I would have to write an ArchR project as output directly after creation of the Arrow files in each workflow version in order to really see the difference in number of cells. I may have to get back to you on this. Thanks for your help!

**Reply by RegnerM2015** [reply]
@rcorces 

To follow up regarding this situation, please see the issues post here:
https://github.com/GreenleafLab/ArchR/issues/724
